File: kyulee/programmers-2/이웃한칸.java
public class 이웃한칸 {
    class Solution {
        public int solution(String[][] board, int h, int w) {
            // 보드의 크기
            int n = board.length;

            // 같은 색으로 칠해진 칸의 개수를 저장할 변수
            int count = 0;

            // h와 w의 변화량을 저장할 배열
            int[] dh = {0, 1, -1, 0};
            int[] dw = {1, 0, 0, -1};

            // 현재 칸의 색깔
            String color = board[h][w];

            // 4방향을 검사
            for (int i = 0; i < 4; i++) {
                int h_check = h + dh[i];
                int w_check = w + dw[i];

                // 인접한 칸이 보드의 범위를 벗어나지 않는지 확인
                if (h_check >= 0 && h_check < n && w_check >= 0 && w_check < n) {
                    // 같은 색이면 count 증가
                    if (board[h][w].equals(board[h_check][w_check])) {
                        count++;
                    }
                }
            }

            return count;
        }
    }
}

File: soyeong/programmers-2/이웃한-칸.py
def solution(board, h, w):
    count = 0

    n = len(board)

    # 좌표 이동
    dx = [1, -1, 0, 0]
    dy = [0, 0, 1, -1]

    # 고른 칸의 색상
    color = board[h][w]

    # 위, 아래, 왼쪽, 오른쪽을 체크
    for i in range(len(dx)):
        # 고른 칸 기준으로 좌표 이동
        nx = h + dx[i]
        ny = w + dy[i]
        # 이동한 칸이 격자보드판을 나가지 않는지 판단
        if 0 <= nx < n and 0 <= ny < n:
            # 해당 색상이 고른 칸이랑 같은지 판단
            if board[nx][ny] == color:
                count += 1
    return count